R/opts.R

```diff
@@ -442,6 +461,98 @@ opt_table_outline <- function(data,
   # Perform input object validation
   stop_if_not_gt(data = data)
 
-  values_vec <- rep(c(style, width, color), 4)
-  tab_options_multi(data, create_option_value_list(table_borders_vec(), values_vec))
+  if (style == "none") {
+    width <- NULL
+    color <- NULL
+  }
+
+  params <-
+    c(style = !is.null(style), width = !is.null(width), color = !is.null(color)) %>%
+    which() %>%
+    names()
+
+  pattern <- paste0("^table\\.border.*?\\.(", paste(params, collapse = "|"), ")")
+
+  # Get vector of `tab_options()` arg names for the table border styles
+  options_vec <- get_tab_options_arg_vec(pattern = pattern)
+
+  if (is.null(width)) width <- NA_character_
+  if (is.null(color)) color <- NA_character_
+
+  values_vec <-
+    dplyr::case_when(
+      grepl("^table\\.border.*?\\.style$", options_vec) ~ as.character(style),
+      grepl("^table\\.border.*?\\.width$", options_vec) ~ as.character(width),
+      grepl("^table\\.border.*?\\.color$", options_vec) ~ as.character(color)
```

Comment:
In the above three lines:
`border.*?` -> `border\\..*?`

Reply:
Now changed.